funfact/lang/_tsrex.py

```diff
@@ -345,10 +345,16 @@ def _getitem(node: _ASNode, indices):  # noqa: F811
     #                         'Indices to a tensor expression must be '
     #                         'abstract indices.'
     #                     )
+    new_indices = tuple([i.root if i is not Ellipsis else Ellipsis for i in
```

Comment:
We can overwrite (rebind) `indices` here since the original one is no longer used after this line, and also since this is a super-small scope.

That said, a bigger concern here is the inclusion of `Ellipse`, which is not of `_ASNode` type, in the list of indices. This could potentially complicate future code development since we will then always need to check if a child is a node or not.

We can probably either:
1. add an ellipsis primitive, or
2. create a special index node.

Option 2 in my opinion may be deceptively simple but could cause problems once users start to decorate the special index with `~` or `*`, etc. :grin: So perhaps option 1 is really the way to go in this case.

Reply:
After a second thought, I am still not sure if a literal Ellipsis or an ellipsis primitive is better. Do you have any further thoughts?

Reply:
I'm looking into this feature again. 

If we go with either option 1 or 2, we have to link the `...` (`Ellipsis`) notation directly with an `_ASNode` type object (be it a primitive or special index node) *before* `_getitem` is called while creating a tensor expression.

Any thoughts on how to achieve this?

Reply:
A solution that I see is to create a function `_as_index` that is called on the indices in the `_getitem` method and replaces an `Ellipsis` with this special index node (or primitive object). 

Would this address the concern you raised above?

> That said, a bigger concern here is the inclusion of Ellipse, which is not of _ASNode type, in the list of indices. This could potentially complicate future code development since we will then always need to check if a child is a node or not.